test/fail_compilation/fail18266.d

```diff
@@ -0,0 +1,28 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail18266.d(22): Error: declaration `fail18266.main.S` is already defined in another scope in `main`
```

Comment:
Would it be complicated to add the line number of where the symbol is first defined to the error message?

Reply:
is this better?

Reply:
Yes, but perhaps as a supplementary error message.

Reply:
I think it's better this way, because the supplemental error message would be too much text with little information

Reply:
This is how the error message would look like with Clang:

```
main.cpp:5:6: error: redefinition of 'foo'
void foo() {}
     ^
main.cpp:3:6: note: previous definition is here
void foo() {}
     ^
1 error generated.
```